tools/build-bpf-skel.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: LGPL-2.1+
+
+import argparse
+import logging
+import pathlib
+import shutil
+import subprocess
+import sys
+
+def clang_arch_flag(arch):
+    return '-D__{}__'.format(arch)
+
+
+def target_triplet():
+    gcc_exec = None
+    try:
+        gcc_exec = shutil.which('gcc')
+    except shutil.Error as e:
+        logging.error('Failed to get which gcc: {}'.format(e))
+        return None
+
+    try:
+        return subprocess.check_output([gcc_exec, '-dumpmachine'],
+                text=True).strip()
+    except subprocess.CalledProcessError as e:
+        logging.error('Failed to get target triplet: {}'.format(e))
+        return None
+
+
+def clang_compile(clang_exec, clang_flags, src_c, out_file, target_arch,
+        target_triplet):
+    clang_args = [clang_exec]
+    clang_args.extend(clang_flags)
+    clang_args.extend([target_arch])
+
+    clang_args.extend([
+        '-I.'])
+
+    if target_triplet:
+        clang_args.extend([
+            '-isystem',
+            '/usr/include/{}'.format(target_triplet)])
+
+    clang_args.extend([
+        '-idirafter',
+        '/usr/local/include',
+        '-idirafter',
+        '/usr/include'])
+
+    clang_args.extend([src_c])
+    clang_args.extend(['-o', out_file])
+
+    logging.debug('Generating ELF object file:')
+    logging.debug('{}'.format(' '.join(clang_args)))
+    subprocess.check_call(clang_args)
+
+
+def llvm_strip(llvm_strip_exec, in_file):
+    llvm_strip_args = [llvm_strip_exec, '-g', in_file]
+
+    logging.debug('Stripping useless DWARF info:')
+    logging.debug('{}'.format(' '.join(llvm_strip_args)))
+
+    subprocess.check_call(llvm_strip_args)
+
+
+def gen_bpf_skeleton(bpftool_exec, in_file, out_fd):
+    bpftool_args = [bpftool_exec]
+    bpftool_args.extend(['g', 's'])
+    bpftool_args.extend([in_file])
+
+    logging.debug('Generating BPF skeleton:')
+    logging.debug('{}'.format(' '.join(bpftool_args)))
```

Comment:
@keszybz

No strong preference, but `logging` has and advantage of passing log level, this may be useful for anyone reusing same infra, e.g. in https://github.com/systemd/systemd/pull/18145 and https://github.com/systemd/systemd/pull/18385.
```
logging.basicConfig(stream=sys.stderr, level=logging.WARNING)
```
logging.WARNING silences debug messages so they don't go to build log.